cmd/finch-daemon/main.go

```diff
@@ -235,3 +241,19 @@ func sdNotify(state string, logger *flog.Logrus) {
 	notified, err := daemon.SdNotify(false, state)
 	logger.Debugf("systemd-notify result: (signaled %t), (err: %v)", notified, err)
 }
+
+// defineDockerConfig defines the DOCKER_CONFIG environment variable for the process
+// to be $HOME/.finch. When building an image via finch-daemon, buildctl uses this variable
+// to load auth configs.
+//
+// This is a hack and should be fixed by passing the actual credentials that come in
+// via the build API to buildctl instead.
+func defineDockerConfig(uid int) error {
```

Comment:
Considering non-VM cases, should this check first if the DOCKER_CONFIG value is already set?

Reply:
This is not a big concern, so can be added as a follow up. And may be irrelevant if we decide to hardcode the environment variable.